checker/tests/lock/LockEffectAnnotations.java

```diff
@@ -141,6 +141,7 @@ void pureMethodWithSynchronizedBlock() {
 
     @GuardedByUnknown class MyClass2 {}
 
+    // :: error: (expression.unparsable.type.invalid)
```

Comment:
I do not understand why this error appears. Can you explain?

Reply:
There is no variable in scope named "lock".
It was a bug before that the Lock Checker did not issue this warning.
Like you, I was surprised to see this bug corrected when I reviewed the code.  But it's an improvement, so I'll take it.

Reply:
The code didn't magically fix a bug. Class declarations don't validate dependent type annotations. I made a proper fix in #2309 and it actually revealed another incorrect test.
This PR only happened to raise an error here because the annotation is copied from the class declaration to the constructor return and constructor return types are validated.
It is much easier to debug these things in the scope of the changes of a PR. Just accepting miraculous improvements leads to much more pain later on.